### PR TITLE
WT-10525 Minimal repro for insert stack race condition

### PR DESCRIPTION
**Pull request by korteland**
Testing, please don't review yet

### CODE REVIEW

**Review by ajmorton — COMMENTED** [review]
Preliminary review as discussed. Mostly nits, but one or two questions around SKIP_PROBABILITY. 
I'm hoping I can make use of `denser_skiplist` in my standalone test. I'll try catch you later today to get your input 👍

File: test/evergreen.yml [hunk 1]
```diff
@@ -3476,6 +3476,40 @@ tasks:
...
+        vars:
+          NONSTANDALONE: -DWT_STANDALONE_BUILD=0
+          CMAKE_TOOLCHAIN_FILE: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v4_gcc.cmake
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/bench/workgen/runner"
+          script: |
+            set -o errexit
+            set -o verbose
+            for i in {1..10}; do ${test_env_vars|} ${python_binary|python3} skiplist_stress.py; done
```
[the remark this answers is not in the document]
Comment (korteland) [reply test/evergreen.yml:3508]: Thanks - your comments on this file are related, I forgot to fix a few things in the standalone test when I changed it so we could make it a real task. the current timeout is fine if you're running it with the "denser skiplists" config. Fixed in d9401d1.

File: src/include/btree_inline.h [hunk 2]
```diff
@@ -1999,9 +1999,16 @@ __wt_page_release(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 static inline u_int
 __wt_skip_choose_depth(WT_SESSION_IMPL *session)
 {
-    u_int d;
+    u_int d, p;
```
[the remark this answers is not in the document]
Comment (korteland) [reply src/include/btree_inline.h]: Thanks, good call - done in d3bd3c7.

File: bench/workgen/runner/skiplist_stress.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from wiredtiger import *
+from workgen import *
+
+context = Context()
+# Connection configuration.
+conn_config = "cache_size=100MB,log=(enabled=false),statistics=[fast],statistics_log=(wait=1,json=false)"
+conn = context.wiredtiger_open("create," + conn_config)
+s = conn.open_session("")
+
+# TODO session config, turn on the skiplist density thing
```
[the remark this answers is not in the document]
Comment (korteland) [reply bench/workgen/runner/skiplist_stress.py:44]: Oh, sorry if that was unclear - it's a comment on how to edit the session config more than it is about enabling it across all sessions. I'm guessing it's in the `conn.open_session("")` call above but I was being lazy when I wrote the comment.

**Review by korteland — COMMENTED** [review]
Sorry about all these comments being in a review, I accidentally clicked "start a review" instead of "add single comment" and there seems to be now way to undo that.

**Pull request closed without merging**